base/linalg/qr.jl

```diff
@@ -471,6 +513,39 @@ Optionally takes a `thin` Boolean argument, which if `true` omits the columns th
 rows of `R` in the QR factorization that are zero. The resulting matrix is the `Q` in a thin
 QR factorization (sometimes called the reduced QR factorization). If `false`, returns a `Q`
 that spans all rows of `R` in its corresponding QR factorization.
+
+# Examples
+```jldoctest
+julia> a = [1. 2.; 3. 4.; 5. 6.];
+
+julia> qra = qrfact(a, Val(true));
+
+julia> full(qra[:Q], thin=true)
```

Comment:
Where are we on the `full(A)` vs `convert(AbstractArray, A)` for things like this? Will this method be eliminated @Sacha0 ?

Reply:
IIRC this method appears in a couple other doctests too. I'm happy to go through and change all of them in a separate PR after this is merged, if that would be ok?

Reply:
There are still there but might go away soon. So far @Sacha0 has mainly eliminated *uses* of `full` in the source code.

Reply:
The blight that is `full` shall one day be ~righteously scoured~ deprecated from this ~earth~ codebase.

Reply:
So do we want to keep this for now or get rid of it?

Reply:
Lets merge while CI is all green. This can be removed together with the rest of the docstring when/if `full` is removed.